rest/specs/src/main/filtered-resources/openapi/openapi.yaml

```diff
@@ -0,0 +1,232 @@
+          application/json:
+            schema:
+              $ref: '#/components/schemas/ErrorMessage'
+    NotImplemented:
+      description: >
+        The requested API version is not supported and therefore not implemented.
+      headers:
+        'Content-Version':
+          $ref: '#/components/headers/ContentVersion'
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/ErrorMessage'
+    BadRequest:
+      description: >
+        The client message could not understood by the server due to invalid format or syntax.
+      headers:
+        'Content-Version':
+          $ref: '#/components/headers/ContentVersion'
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/ErrorMessage'
+    DefaultError:
+      description: >
+        Any other possible errors not currently known.
+      headers:
+        'Content-Version':
+          $ref: '#/components/headers/ContentVersion'
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/ErrorMessage'
+  parameters:
+    Accept-Version:
+      name: Accept-Version
+      in: header
+      description: >
+        The minimal API version that a client using this API will accept
+        responses from.
+      required: true
+      schema:
+        $ref: '#/components/schemas/Version'
+  headers:
+    ContentVersion:
+      description: >
+        The API version used by the server to produce the REST message. The server will accept
+        messages for any minor versions prior to this one.
+      required: true
+      schema:
+        $ref: '#/components/schemas/Version'
+      examples:
+        major.minor:
+          value: '1.2'
+          summary: A released minor version
+        major.minor.patch:
+          value: '1.0.3'
+          summary: A patched version
+        dev:
+          value: '1.2.0-SNAPSHOT'
+          summary: A development version
+  schemas:
+    Product:
+      description: The JSON description of the Product being ingested.
+      type: object
+      properties:
+
+        productInfo:
+          $ref: '#/components/schemas/ProductInfo'
+        file:
+          $ref: '#/components/schemas/File'
+      required:
+        - productInfo
+        - file
+    ProductInfo:
+        properties:
+          title:
+            type: string
+            minLength: 1
+            description: The title of the Product.
+            example: A Picture
+          filename:
+            type: string
+            minLength: 1
+            description: The name of the source file.
+            example: product.jpeg
+          bytes:
+            type: integer
+            description: The size of the file payload.
+            format: int32
+            minimum: 0
+            exclusiveMinimum: true
+            example: 1024
+          mimeType:
+            type: string
+            description: The MIME type of the file being sent.
```

Comment:
❓Should we have clearer constraints here? For instance, what if the client uses `application/octet-stream`, is the service going to accept or reject the request?

Reply:
a regex for the definition of a mime would be a good start I think. Then I agree with @lessarderic, if we don't plan on supporting all possible mimes then we need to restrict that even more.

Reply:
I think regex constraints are good enough for now. Unless we have a defined list of mime types that we will support, I don't think we should start restricting it just yet.

Reply:
agreed. A regex to mimes should be good enough right now